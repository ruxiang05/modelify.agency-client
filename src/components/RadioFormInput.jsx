import React from 'react';
import PropTypes from 'prop-types';

const RadioFormInput = ({
  title, name, value, handleChange, id, checked,
}) => (
  <React.Fragment>
    <input
      id={id}
      name={name}
      type="radio"
      value={value}
      onChange={handleChange}
      checked={checked}
    />
    <label htmlFor={id}>{title}</label>
  </React.Fragment>
);

RadioFormInput.propTypes = {
  id: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  checked: PropTypes.bool,
};

RadioFormInput.defaultProps = {
  checked: false,
};
export default RadioFormInput;
