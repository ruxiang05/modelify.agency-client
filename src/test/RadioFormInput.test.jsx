/* Uses methods and/or components from react and enzyme */
import React from 'react';
import { shallow } from 'enzyme';
import RadioFormInput from '../components/RadioFormInput';

const input = {
  id: '1',
  name: 'name',
  value: '',
  handleChange: jest.fn(),
  placeholder: 'placeholder',
};

describe('RadioFormInput', () => {
  it('should render RadioFormInput', () => {
    const wrapper = shallow(<RadioFormInput {...input} />);
    expect(wrapper).toMatchSnapshot();
  });
});
