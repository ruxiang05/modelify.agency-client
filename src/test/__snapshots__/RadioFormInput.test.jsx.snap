// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioFormInput should render RadioFormInput 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioFormInput
    checked={false}
    handleChange={[MockFunction]}
    id="1"
    name="name"
    placeholder="placeholder"
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          checked={false}
          id="1"
          name="name"
          onChange={[MockFunction]}
          type="radio"
          value=""
        />,
        <span
          className="radio"
        />,
        <label
          htmlFor="1"
        />,
      ],
      "className": "radio-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": false,
          "id": "1",
          "name": "name",
          "onChange": [MockFunction],
          "type": "radio",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "radio",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "htmlFor": "1",
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            checked={false}
            id="1"
            name="name"
            onChange={[MockFunction]}
            type="radio"
            value=""
          />,
          <span
            className="radio"
          />,
          <label
            htmlFor="1"
          />,
        ],
        "className": "radio-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": false,
            "id": "1",
            "name": "name",
            "onChange": [MockFunction],
            "type": "radio",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "radio",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "htmlFor": "1",
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
